src/domain/event/occurrences/OccurrenceInfo.tsx

```diff
@@ -181,14 +184,15 @@ const OccurrenceInfo: React.FC<{
         language={locale}
         variant="button"
       />
-      {hasExternalEnrolment ? (
+      {hasExternalEnrolment && (
```

Comment:
Seems like you got rid of nested ternary which is fine for me but I think only one of these buttons or only the enrolment start notice should be rendered at a time 🤔

Reply:
This might also need checks if enrolment has ended. We had that previously but seems like I forgot to add those there during the redesign.

Seems like old component `EnrolmentButtonCell` is still in the project. You could take a look how enrolment button and other notices were rendered previously.

Reply:
> Seems like you got rid of nested ternary which is fine for me but I think only one of these buttons or only the enrolment start notice should be rendered at a time 🤔

Doesn't it work in this way in this changeset? Buttons and notices are not shown at the same time.

Do you mean that external enrolments also require a notice?

Reply:
I think it does work correctly but having `!hasExternalEnrolment && neededOccurrences === 1 && !isEnrollmentOpen` and `hasExternalEnrolment` as separate checks makes this part of the code hard to understand

Reply:
Ah, now I maybe understood you. Do you have something in mind to fix it with?

This part of the logic in the component is hard to reason about, but I feel like these changes haven't increased or decreased it by much. Not chaining elvis is something people consistently recommend so it seemed like a good low hanging fruit to pick.

The next things I'd be looking at is to formalise the rendering in some way--perhaps by having a state string that describes the button that should be rendered. But imo that's a bit heavy for now.

Now that we are handling the errors in this PR as well, this part may need reafctoring.

Reply:
I think the easiest way to make this easier to reason about would be to have function/Component that just has if statements with early returns like `EnrolmentButtonCell.tsx` did previously. That way it is pretty clear that only one thing is rendered at a time

Reply:
Which would be some kind of notice or enrolment button of 2 different kinds

Reply:
This got refactored while I added the error handling.